ui/apps/platform/src/Containers/MainPage/InviteUsersModal.tsx

```diff
@@ -1,21 +1,126 @@
+    emails: yup
+        .string()
+        .transform((value) => dedupeDelimtedString(value).join(',')) // dedupe - optional step
+        .required('At least one email is required')
+        .test('emails', 'Invalid email address', (value) =>
+            Boolean(
+                value &&
+                    value
+                        .split(',')
+                        .map((v) => v.trim())
+                        .every(isEmail)
+            )
+        ), // .string().trim().required('At least one email is required'),
+    provider: yup.string().required('An auth provider is required'),
+    role: yup.string().required('A role is required'),
+});
+
+const defaultInviteFormValues: InviteFormValues = {
+    emails: '',
+    provider: '',
+    role: '',
+};
 
 const feedbackState = createStructuredSelector({
+    authProviders: selectors.getAvailableAuthProviders,
+    groups: selectors.getRuleGroups,
+    roles: selectors.getRoles,
+    isFetchingAuthProviders: (state) =>
+        selectors.getLoadingStatus(state, authActionTypes.FETCH_AUTH_PROVIDERS) as boolean,
+    isFetchingRoles: (state) =>
+        selectors.getLoadingStatus(state, roleActionTypes.FETCH_ROLES) as boolean,
     invite: selectors.inviteSelector,
 });
 
 function InviteUsersModal(): ReactElement | null {
     const { invite: showInviteModal } = useSelector(feedbackState);
     const dispatch = useDispatch();
+    const formik = useFormik<InviteFormValues>({
+        initialValues: defaultInviteFormValues,
+        onSubmit: () => {}, // required but not used, because the submit action is in the modal footer
+        validationSchema,
+        validateOnMount: true,
+    });
+
+    const { isValid, values, setFieldValue } = formik;
+
+    const { authProviders, groups, roles } = useSelector(feedbackState);
+
+    const authProvidersWithRules = mergeGroupsWithAuthProviders(authProviders, groups);
+
+    const allowedRoles = roles.filter((role) => {
+        return role.name !== 'Admin' && role.name !== 'None';
+    });
+
+    useEffect(() => {
+        dispatch(authActions.fetchAuthProviders.request());
+        dispatch(roleActions.fetchRoles.request());
+        dispatch(groupActions.fetchGroups.request());
+    }, [dispatch]);
+
+    useEffect(() => {
+        const typedProviders = authProviders as AuthProvider[]; // redux state from sagas is not yet typed
+
+        // if there is only 1 authProvider, pre-select it in the dropdown
+        if (typedProviders.length === 1) {
+            // eslint-disable-next-line no-void
+            void setFieldValue('provider', typedProviders[0].name);
+        } else if (typedProviders.length > 1) {
+            // if there is more than 1 authProvider, pre-select RedHat / OpenShift provider if possible
+            const redhatSsoProvider = typedProviders.find(
+                (provider) =>
+                    provider.name.includes('Red Hat SSO') || provider.name.includes('Red Hat SSO')
```

Comment:
Is the second includes here supposed to be OpenShift related?

Reply:
Yes, should be "OpenShift".

This also reminds me to make the checks case-insensitive.